vignettes/speedup_options.Rmd

```diff
@@ -0,0 +1,710 @@
+    rt = rt_opts(
+      prior = rt_prior_default,
+      rw = 7
+    )
+  ),
+  # Model that uses the variational inference method instead of MCMC (from rstan) 
+  default_vb_rstan = list(
+    stan = stan_opts(
+      method = "vb"
+    )
+  ),
+  # Model that uses the variational inference method instead of MCMC (from cmdstanr) 
+  default_vb_cmdstanr = list(
+    stan = stan_opts(
+      method = "vb",
+      backend = "cmdstanr"
+    )
+  )
+)
+```
+
+## Running the models
+
+Let's run the models and gather the results.
+
+Let's combine the shared model inputs into a list for use across all the models.
+
+``` r
+model_inputs <- list(
+  data = reported_cases_true[1:(.N - horizon),],
+  generation_time = generation_time_opts(generation_time),
+  delays = delay_opts(delay),
+  obs = obs,
+  horizon = horizon,
+  verbose = FALSE
+)
+```
+
+To run the models, we will sweep across the list of models `models` and shared model inputs `model_inputs`. We will use the `epinow()` function and return useful outputs like the timing of model runs.
+
+Note that we will run the model using 133 observations so that we can compare the forecast performance over the full set of observations.
+
+``` r
+# Create a version of epinow() that works like base::try() and works even if some models fail.
+safe_epinow <- purrr::safely(epinow)
+# Run the models
+results <- lapply(
+  model_configs,
+  function(model) {
+    do.call(
+      safe_epinow,
+      c(
+        model_inputs,
+        model
+      )
+    )
+  }
+)
+```
+
+## Evaluating performance
+
+### Run times
+
+Let's see how long each model took to run.
+
+
+```{.r .fold-hide}
+# Extract the run times
+runtimes <- lapply(
+  results,
+  function(x) {
+    if (is.null(x$error)) {
+      return(round(as.duration(x$result$timing), 1))
+    } else {
+      return(as.duration("NA"))
+    }
+  }
+)
+
+# Convert to a table format
+
+runtimes_dt <- melt(
+  as.data.table(runtimes),
+  measure.vars = names(model_configs),
+  variable.name = "model",
+  value.name = "runtime"
+)
+
+# Add model descriptions
+
+runtimes_dt <- merge(
+  runtimes_dt,
+  model_descriptions,
+  by = "model"
+)
+
+setcolorder(runtimes_dt, c("model", "description", "runtime"))
+
+# Order by run time
+
+runtimes_dt <- runtimes_dt[order(runtime), ]
+ 
+# Print table
+knitr::kable(runtimes_dt[, c("model", "description", "runtime")], caption = "Run times for various _EpiNow2_ models")
+```
+
+
+
+Table: Run times for various _EpiNow2_ models
+
+|model               |description                                                                                                            |                  runtime|
+|:-------------------|:----------------------------------------------------------------------------------------------------------------------|------------------------:|
+|default_vb_rstan    |Default model; Variational algorithm for sampling instead of MCMC and using [`{rstan}`]                                |                     3.7s|
```

Comment:
As I said in a previous review round I think its useful to distinguish in the output between model differences and fitting approaches differences. What do you think about that?